docs/design/001-cli-config-structure.md

```diff
@@ -0,0 +1,115 @@
+- can every option in the config file be passed to the cli?
+
+The default behavior should be consistent with how bazel interacts with bazelrc files. As those using the aspect cli could be familiar with how bazel interacts with it's rc files already.
+
+This document should provide enough information so that so that someone should be able to either understand how to setup a configuration file so that they can enable the plugins and features they would like to use.
+
+By reading this document someone should be able to either understand how to operate a configuration file so that they can setup the plugins and features of those plugins that they would like to use. Or they should understand the configuration file that their plugin will need to consume and the format that it should use to be consistent with all other plugins.
+
+Non-goals:
+
+- how these choices should be implemented
+
+Currently there is no agreed upon way for the config files to be laid out and for their precedence to be understood. The goal of one section of this document it to make it clear what configurations files are available, how they interact with one another, how they can be overridden and how they can be changed.
+
+The goal of the second section of this document is to establish a clear format, expectation and language around the structure of the config file and its contents along with how that maps to plugins and features of those plugins.
+
+## Proposal
+
+First lets create some common definitions to remove ambiguity surrounding language with regards to the aspect cli.
+
+## Definitions
+
+### Plugin
+
+For the aspect cli a plugin refers to something that exposes a grpc server this will typically be a binary.
+The definition of a plugin is not associated with any functionality inside that binary or anything else but simply that it exposes a grpc server and interacts though that server.
+
+Plugins can be turned on and off independently unless they are explicitly specified as a dependency of another plugin. This will be discussed further later on.
+
+### Feature
+
+A feature refers to some functionality that is exposed through a plugin. There can be multiple features in a single plugin.
+
+Features should be exposed independently inside of a plugin so that they can be turned on and off independently.
+
+## Config File Content Structure
+
+The aspect cli config should be broken down into different namespaces that map to each plugin. This structure can be nested so that features of plugins can also use their own namespace structure to control their own features and settings. The core aspect cli will only deal with the top level namespaces and turing those plugins on appropriately. Dealing with the rest is up to the creator of the plugin. It is recommended that the same namespace structure is used to control features of a plugin as well so that it is easy and consistent for users to understand a config file with multiple plugins and update them as needed.
```

Comment:
This feels like a really early draft. What are these namespaces? I think it's hard to discuss the structure without a little example here.